vassal-app/src/main/java/VASSAL/tools/ArchiveWriter.java

```diff
@@ -197,6 +197,20 @@ public void removeImage(String name) {
     localImages = null;
   }
 
+  /**
+   * Removes a file in the archive
+   * @param name file in the archive to be removed
+   */
+  public void removeFile(String name) {
+    try {
+      archive.remove (name);
+    }
+    catch (IOException e) {
+      WriteErrorDialog.error(e, archive.getName());
+    }
+  }
```

Comment:
While you're at it, you could replace all but the last line of `removeImage()` with a call to `removeFile(imageDir + name)`.

Reply:
_all but the last line_. You killed the last line.